src/AutoRest.CSharp/Common/Generation/Writers/SerializationWriter.cs

```diff
@@ -201,9 +201,16 @@ private void WriteJsonDeserialize(CodeWriter writer, SchemaObjectType model, Jso
                     }
                 }
 
-                writer.DeserializeValue(jsonSerialization,
-                    w=> w.AppendRaw("element"),
-                    (w, v) => w.Line($"return {v};"));
+                if (model.Declaration.IsAbstract)
+                {
+                    writer.Line($"throw new {typeof(Exception)}();");
```

Comment:
We probably should have some text for exception here.

Reply:
flip it to the `NotSupportedException` . @chamons do you have ideas for a better message?

Reply:
`throw new NotSupportedException ($"Deserialization of abstract type '{model.Type}' not supported.");`

Reply:
Fixed